azurerm/internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -118,6 +118,19 @@ func dataSourceKubernetesCluster() *schema.Resource {
 								},
 							},
 						},
+
+						"open_service_mesh": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+								},
+							},
+						},
```

Comment:
we'll also need to flatten this value for the Data Source, this is done in `flattenKubernetesClusterDataSourceAddonProfiles` (by parsing the add-on profile from the response and reading the enabled state)